ppr-api/src/ppr_api/models/registration.py

```diff
@@ -910,10 +910,29 @@ def get_former_party_name(self, new_party: Party):
                     elif party.business_name:
                         former_name = party.business_name
                     else:
-                        former_name = party.last_name + ', ' + party.first_name
-                        if party.middle_initial:
-                            former_name += ' ' + party.middle_initial
-                    return former_name
+                        # match if only 1 name is different in addition to same address.
+                        former_name = self.__get_matching_party_name(new_party, party)
+                    if former_name:
+                        return former_name
+        return former_name
```

Comment:
Are there any instances where the first name and last name change?

Reply:
Then it's not a name change: cannot distinguish between a different person at the same address.